### PR TITLE
redis: conn pool stats and connect timeouts

### PR DESCRIPTION
**Pull request by mattklein123**

### CODE REVIEW

**Comment by mattklein123** [comment]
@lyft/network-team @danielhochman

File: test/run_envoy_coverage.sh [hunk 1]
```diff
@@ -6,9 +6,10 @@ PROJECT_SOURCE_DIR=$3
 PROJECT_BINARY_DIR=$4
 GCOVR=$5
 GCOVR_EXTRA_ARGS=$6
+EXTRA_SETUP_SCRIPT=$7
```
Comment (RomanDzhabarov) [inline test/run_envoy_coverage.sh]: what is this for? seems to sneak into this pr.

File: test/common/redis/conn_pool_impl_test.cc [hunk 2]
```diff
@@ -129,13 +167,44 @@ TEST_F(RedisClientImplTest, FailAll) {
...
+TEST_F(RedisClientImplTest, FailAllWithCancel) {
+  InSequence s;
+
+  setup();
+  onConnected();
+
+  Network::MockConnectionCallbacks connection_callbacks;
+  client_->addConnectionCallbacks(connection_callbacks);
+
+  RespValue request1;
```
Comment (RomanDzhabarov) [inline test/common/redis/conn_pool_impl_test.cc]: nit: is there any significance in the name1 in those tests?

File: test/run_envoy_coverage.sh (hunk 1 above)
Comment (mattklein123) [inline test/run_envoy_coverage.sh]: Coverage is broken on our private build without this change, I just fixed it here. It's related to other test PRs that @htuch has been doing.

File: test/common/redis/conn_pool_impl_test.cc (hunk 2 above)
Comment (mattklein123) [inline test/common/redis/conn_pool_impl_test.cc]: I might clean this up in a subsequent PR but I don't think this is worth taking the time to change right now.

Comment (RomanDzhabarov) [inline test/common/redis/conn_pool_impl_test.cc]: yeah, that's fine to keep this review clean of refactoring/renaming

**Pull request merged**